### PR TITLE
Components file as source of truth + tooling

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This brings a new configuration file, on the root, to serve as the
source of truth for component version for the operator. Any script
should use this file as a source of truth.

In addition, this bring the start of a tool to quickly get the version
of each component (example of usage in `./test/config.sh`) as well as
checking if there is new version of the component (`check` and
`check-bugfix`).

The next step here is to enhance this tool to be able to update the
`components.yaml` on its own (depending if we want to bump to the next
bugfix release or the next minor/major one). From there, we will be
able to automate the updates of those, a bit à-la-dependabot.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

This takes over https://github.com/tektoncd/operator/pull/641 in a simpler way as it doesn't touch anything related to fetching the payload.

/kind misc

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: components.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  version: v0.9.0
+results:
+  github: tektoncd/results
+  version: v0.4.0
+hub:
+  github: tektoncd/hub
+  version: v1.8.0
+pipelines-as-code:
+  github: openshift-pipelines/pipelines-as-code
+  version: 0.10.1
```
Comment (vdemeester) [inline components.yaml:26]: Note: this doesn't support `stable` or `latest` or `nightly` 🙃 The idea behind it is really to do something similar to what `dependabot` does: we always use released version, and an automated tooling is looking at new releases and propose PRs to bump to those.

**Comment by sm43** [comment]
this is for figuring out the latest version right?
and fetch release script will still be used to fetch the relase yamls?
don't we need to similar changes in `makefile` as did in `test/config.sh`?

**Comment by vdemeester** [comment]
> this is for figuring out the latest version right? and fetch release script will still be used to fetch the relase yamls? don't we need to similar changes in `makefile` as did in `test/config.sh`?

@sm43 Yes, this is to "figure out the latest version" (with some rules) and to update the version in case of. And yes, the fetch release scripts will still be used. Yes we need, I forgot there was something in the `Makefile` 😝

**Comment by vdemeester** [comment]
Note that fetch-script will be updated long term to read the `components.yaml` instead of relying on environment variables.

**Comment by vdemeester** [comment]
@sm43 well, I ended up adapting `fetch-release.sh` to read the components file.

**Comment by vdemeester** [comment]
https://github.com/tektoncd/plumbing/pull/1134 to build a "nightly"

**Review by concaf — COMMENTED** [review]
This is awesome! 🎉 
Can we also make CSV template use the components.yaml as well?

File: cmd/operator-tool/check.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	if err != nil {
+		return err
+	}
+	g, ctx := errgroup.WithContext(context.Background())
+	for name, component := range components {
+		// Force scope
+		name := name
+		component := component
+
+		g.Go(func() error {
```
Comment (pradeepitm12) [inline cmd/operator-tool/check.go:39]: :+1:

**Pull request merged**